src/vos/vos_common.c

```diff
@@ -352,6 +352,18 @@ vos_tls_init(const struct dss_thread_local_storage *dtls,
 		return NULL;
 	}
 
+	rc = d_dtm_init(&tls->vtl_dtm, NULL);
+	if (rc) {
+		vos_ts_table_free(&tls->vtl_ts_table);
+		umem_fini_txd(&tls->vtl_txd);
+		vos_imem_strts_destroy(&tls->vtl_imems_inst);
+		D_FREE(tls);
+		return NULL;
+	}
```

Comment:
Happy to see it per xstream. Shouldn't we change DTM to have no locking too in this case since this is single consumer? We really don't need the dtm->dtm_lock in this case.